### PR TITLE
API Overhaul

### PR DESCRIPTION
**Pull request by RlonRyan**
Rework API as to make it more logical and clean. There is still quite a bit of work to be done, but I would like some confirmation that I am on the right track...

### CODE REVIEW

File: src/main/java/com/infinityraider/agricraft/api/example/ExampleAgriCraftAPIimplementation.java [hunk 1]
```diff
@@ -2,10 +2,10 @@
 
 import com.infinityraider.agricraft.api.API;
 import com.infinityraider.agricraft.api.APIBase;
-import com.infinityraider.agricraft.api.v1.APIv1;
+import com.infinityraider.agricraft.api.v3.APIv3;
```
Comment (InfinityRaider) [inline src/main/java/com/infinityraider/agricraft/api/example/ExampleAgriCraftAPIimplementation.java]: Whats the point of v3? Its a new version of mc, might as well reset api version.

Comment (RlonRyan) [inline src/main/java/com/infinityraider/agricraft/api/example/ExampleAgriCraftAPIimplementation.java]: That's true...

**Comment by RlonRyan** [comment]
I guess I was just thinking in terms of continuity, but I like the idea of keeping the version number low too.

**Comment by RlonRyan** [comment]
Also... I saw you made a repository for the API... Are we planning to move the API there?

**Comment by InfinityRaider** [comment]
I was goint to do that, but I couldn't get it to work.

**Comment by RlonRyan** [comment]
You want me to try and move the API for you?

**Comment by InfinityRaider** [comment]
Thanks, but there is no need for it.

**Comment by RlonRyan** [comment]
If you are sure.

**Comment by RlonRyan** [comment]
You think I could maybye merge these soon? I fear they are getting close to being runaway branches, and I wouldn't want to waste work on something that won't even be used...

**Comment by InfinityRaider** [comment]
Sure, you can merge it.

**Pull request merged**